jetbrains-core/resources/META-INF/plugin.xml

```diff
@@ -224,6 +224,14 @@
 
         <webHelpProvider implementation="software.aws.toolkits.jetbrains.core.help.HelpIdTranslator"/>
         <fileEditorProvider implementation="software.aws.toolkits.jetbrains.services.s3.S3ViewerEditorProvider"/>
+
+        <!-- CloudFormation template linting -->
+        <externalAnnotator language="JSON"
+                           id="annotatorJSON"
+                           implementationClass="software.aws.toolkits.jetbrains.services.cloudformation.annotations.ErrorAnnotator"/>
+        <externalAnnotator language="yaml"
+                           id="annotatorYAML"
+                           implementationClass="software.aws.toolkits.jetbrains.services.cloudformation.annotations.ErrorAnnotator"/>
```

Comment:
Does this mean [`cfn-lint`](https://github.com/aws-cloudformation/cfn-python-lint/) is run on all `.json`/`.yaml` files?
[Could borrow some CloudFormation template recognition logic from the Visual Studio Code extension to check for either `AWSTemplateFormatVersion ` or `Resources` and `Type`](https://github.com/aws-cloudformation/aws-cfn-lint-visual-studio-code/blob/edd51fb86441a1c48d294c4b664189c6f4d11399/server/src/server.ts#L125)

Reply:
Yes, right now this code runs on all json and yaml files.

We could run some checks beforehand but I'm afraid this will have performance implications, as the plugin would have to read the file every time it changed before sending it over to the linter.

Maybe it would be better to have the linter return as fast as possible once it realises that the file isn't a template. Plus this means that we don't have duplicate checks both here and in cfn-lint.

Reply:
The downsides of not using a required marker or file extension :/

Should some form of warning be added to cfn-lint push people back to a better path by specifying AWSTemplateFormatVersion?

This right now will invoke the process for every JSON/YAML file they open (and when they type in them as well)

Merging in support for CFN as a file type would act as the gate in the future though.

Reply:
I would personally prefer making a file type, [here's how JB CF plugin does it](https://github.com/shalupov/idea-cloudformation/blob/3195bad973e5b1b4b0913d41d2eb9b6f099ea9cc/src/main/kotlin/com/intellij/aws/cloudformation/CloudFormationFileTypeDetector.kt) . The detection heuristic can probably be improved.

Reply:
I have updated the code so that the linter only runs if `AWSTemplateFormatVersion` or `Resources` are found.